backend/src/v1/controllers/wordController.ts

```diff
@@ -1,31 +1,93 @@
 import { Request, Response } from "express";
+import { ProvideWordRepository, WordRepository } from "../data/repositories/wordRepository";
 
-interface WordController {
-    getWords: (req: Request, res: Response) => void;
-    getWord: (req: Request, res: Response) => void;
-    addWord: (req: Request, res: Response) => void;
-    updateWord: (req: Request, res: Response) => void;
-    deleteWord: (req: Request, res: Response) => void;
+export interface WordController {
+    getWords: (req: Request, res: Response) => Promise<void>;
+    getWord: (req: Request, res: Response) => Promise<void>;
+    addWord: (req: Request, res: Response) => Promise<void>;
+    updateWord: (req: Request, res: Response) => Promise<void>;
+    deleteWord: (req: Request, res: Response) => Promise<void>;
 };
 
-export function WordController(): WordController {
-    return WordControllerImpl;
+export function ProvideWordController(): WordController {
+    return new WordControllerImpl(ProvideWordRepository());
 }
 
-const WordControllerImpl: WordController = {
-    getWords: function (req: Request, res: Response): void {
-        res.send("Get all words");
-    },
-    getWord: function (req: Request, res: Response): void {
-        res.send("Get word by ID");
-    },
-    addWord: function (req: Request, res: Response): void {
-        res.send("Add a new word");
-    },
-    updateWord: function (req: Request, res: Response): void {
-        res.send("Update an existing word");
-    },
-    deleteWord: function (req: Request, res: Response): void {
-        res.send("Delete an existing word");
+class WordControllerImpl implements WordController {
+
+    repository: WordRepository = ProvideWordRepository();
+    constructor(repository: WordRepository) {
+        this.repository = repository;
     }
+
+    getWords = async (req: Request, res: Response) => {
+        let words = await this.repository.getWords();
+        res.json(words);
+    };
+    getWord = async (req: Request, res: Response) => {
+        try {
+            const id = parseInt(req.params.id);
```

Comment:
Always pass the second parameter to `parseInt`:

```ts
const id = parseInt(req.params.id, 10);
```

There's an eslint rule for this that explains why https://eslint.org/docs/latest/rules/radix